src/Entity/Donation.php

```diff
@@ -279,10 +246,58 @@ public function nextDonationAt(\DateTime $fromDay = null): \DateTime
             $fromDay = new \DateTime();
         }
 
-        $donationDate = clone $this->donatedAt;
+        $donationDate = clone $this->createdAt;
 
         return $donationDate->modify(
             sprintf('+%d months', $donationDate->diff($fromDay)->m + 1)
         );
     }
+
+    /**
+     * @return Collection|Transaction[]
+     */
+    public function getTransactions(): Collection
+    {
+        return $this->transactions;
+    }
+
+    public function getStatus(): string
+    {
+        return $this->status;
+    }
+
+    public function hasError(): bool
+    {
+        return self::STATUS_ERROR === $this->getStatus();
+    }
+
+    public function isFinished(): bool
+    {
+        return self::STATUS_FINISHED === $this->getStatus();
+    }
+
+    public function isSubscriptionInProgress(): bool
+    {
+        return self::STATUS_SUBSCRIPTION_IN_PROGRESS === $this->getStatus();
+    }
+
+    public function isSuccess(): bool
```

Comment:
Why do we keep this method ? I don't see how a donation can be successful when it's finished or in progress. 

IMHO, it does not make sense anymore.

Reply:
It's used in result page when the user come back to the website to display success page or error page

Reply:
How is this accurate ? What's the workflow of these statuses ?

Reply:
I am sorry but the semantic is wrong somewhere.

From a human point of view: neither `finished` nor  `subscription_in_progress`  implies success.

Any thoughts @ottaviano @Remg @thomasbisignani @illusionOfParadise ?